components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp

```diff
@@ -343,7 +344,7 @@
                                     <td colspan="3">
                                         <a class="icon-link"
                                            style="background-image:url(../admin/images/add.gif);"
-                                           href='addOperationParameter.jsp?operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=action%>&queryId=<%=Encode.forHtmlAttribute(selectedQueryId)%>' ><fmt:message
+                                           href='addOperationParameter.jsp?operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=Encode.forHtml(action)%>&queryId=<%=Encode.forHtmlAttribute(selectedQueryId)%>' ><fmt:message
```

Comment:
forUriComponent should be used
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String-